core/vpgl/file_formats/vpgl_geo_camera.h

```diff
@@ -156,6 +164,8 @@ class vpgl_geo_camera : public vpgl_camera<double>
 
   vnl_matrix<double>  trans_matrix(){return trans_matrix_; }
 
+  //: since vpgl_geo_camera is not templated only vpgl_camera<double>* is covariant with vpgl_camera<T>*
+  virtual vpgl_camera<double>* clone(void) const {return new vpgl_geo_camera(*this);}
```

Comment:
Why can't this clone method return a vpgl_geo_camera* ?  I don't understand the explanation in the comment;  A vpgl_geo_camera* can be safely downcasted to a vpgl_camera<double>*  (For what its worth, it compiles fine for me on linux (gcc) and mac (clang) )